### ISSUE TITLE
"IbAutoClosedown=no" not working 

### ISSUE THREAD

**Issue by luculli** [ib-controller/ib-controller#92]
Hi,

I just tried ib-controller with TWS 956 and the option "IbAutoClosedown=no" doesn't seem to work because TWS shutdown automatically ...

Thanks,

**Comment by luculli** [comment]
I just tried again and it generally works. However, when the auto log off time is set just 1 minute (or less) than the actual time, it doesn't seem to work (this is what I did to test it before writing my previous comment).

So, no problem! It's Ok in practice.

**Comment by rlktradewright** [comment]
This is because TWS puts up the Exit Session dialog, which notifies the user that it's about to shut down, 5 minutes before it actually does shut down - only when it does this can IBController intervene and r-set the autologoff time. But if the autologoof time is set very close to the current time, TWS just assumes that you want out and doesn't display the Exit Session at all, so IBController never knows about it, and TWS just shuts down.

**Comment by SDiogo** [comment]
Hi, I am quite new with IBController; however, I am facing the same issue and I am not understanding what was mentioned here regarding "But if the autologoof time is set very close to the current time, TWS just assumes that you want out and doesn't display the Exit Session at all, so IBController never knows about it, and TWS just shuts down."

So, on TWS, I can configure the autologoff through "Set Auto Log Off Time (HH:MM)" - let's say it is defined to 01:15 AM. What needs to be done at IBController side to avoid it to let TWS shutdown?

Thanks,
SDiogo

**Comment by rlktradewright** [comment]
All you have to do is set:

`IbAutoClosedown=no`

The point of the comment you were referring to, is that if you've set the autologoff time in TWS to 01:15 AM (as you suggest)), and you've set `IbAutoClosedown=no`, and then you start IBController at say 01:12 AM (ie less than 5 minutes before the autlogoff time), then TWS will exit at 01:15 AM because it doesn't display the Exit Session dialog.

In other words, if you want to test whether `IbAutoClosedown=no` works (and I can absolutely categorically assure you that it does), then make sure you start it at least 5 minutes before TWS's autologoff time.

As a corollary to this, if you start IBController automatically (eg via a scheduled task or cron), make sure the TWS autologoff time is not set just after the time you start IBController.
